editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -382,11 +382,11 @@ impl MessageHandler<DocumentMessage, DocumentMessageData<'_>> for DocumentMessag
 
 					let name = self.network_interface.frontend_display_name(&layer.to_node(), &[]);
 
-					let (scale, angle, translation) = self.metadata().document_to_viewport.to_scale_angle_translation();
-					let translation = translation + scale * bounds[0].min(bounds[1]) - DVec2::Y * 4.;
-					let transform = DAffine2::from_angle_translation(angle, translation);
+					let transform = self.metadata().document_to_viewport
+						* DAffine2::from_scale(DVec2::splat(self.document_ptz.zoom().recip()))
+						* DAffine2::from_translation(bounds[0].min(bounds[1]) - DVec2::Y * 4.);
 
-					overlay_context.text_with_transform(&name, COLOR_OVERLAY_GRAY, None, transform, Pivot::BottomLeft);
+					overlay_context.text(&name, COLOR_OVERLAY_GRAY, None, transform, 0., [Pivot::Start, Pivot::End]);
```

Comment:
This matrix math isn't quite working: the text flies off when the artboard is moved away from the origin. But it looks like it's working stably when panned and zoomed, so it's just the artboard offset causing this.